packages/orchestrator-ui-components/src/components/WfoBadges/WfoWorkflowTargetBadge/WfoWorkflowTargetBadge.tsx

```diff
@@ -0,0 +1,66 @@
+import { WfoBadge } from '../WfoBadge';
+import React, { FC } from 'react';
+import { useOrchestratorTheme } from '../../../hooks';
+import { WorkflowTarget } from '../../../types';
+
+export type WfoWorkflowTargetBadgeProps = {
+    target: WorkflowTarget;
+};
+
+export const WfoWorkflowTargetBadge: FC<WfoWorkflowTargetBadgeProps> = ({
+    target,
+}) => {
+    const { theme, toSecondaryColor } = useOrchestratorTheme();
+
+    const getBadgeColorFromTarget = (_target: string) => {
+        const {
+            primary,
+            danger,
+            dangerText,
+            primaryText,
+            success,
+            successText,
+            warning,
+            warningText,
+        } = theme.colors;
+
+        switch (_target?.toLowerCase()) {
+            case WorkflowTarget.CREATE:
+                return {
+                    badgeColor: toSecondaryColor(success),
+                    textColor: successText,
+                };
+            case WorkflowTarget.MODIFY:
+                return {
+                    badgeColor: toSecondaryColor(primary),
+                    textColor: primaryText,
+                };
+            case WorkflowTarget.SYSTEM:
+                return {
+                    badgeColor: toSecondaryColor(warning),
+                    textColor: warningText,
+                };
+            case WorkflowTarget.TERMINATE:
+                return {
+                    badgeColor: toSecondaryColor(danger),
+                    textColor: dangerText,
+                };
+            default:
+                return {
+                    badgeColor: toSecondaryColor(primary),
+                    textColor: primaryText,
+                };
+        }
+    };
+
+    const { badgeColor, textColor } = getBadgeColorFromTarget(target);
+
+    if (target !== null) {
```

Comment:
Small style preference. I would to the nullCheck on the target on line 13 so you bail out of the function immediately when there is not target. So 

```if(!target) return null```

Reply:
Thanks, I updated code.